src/backend/Csrs.Api/Services/FileService.cs

```diff
@@ -65,46 +77,54 @@ public class FileService : IFileService
             }
             catch (Exception ex)
             {
-                _logger.LogError(ex, "Exception in FileService.CreateFile");
+                _logger.LogError(ex, "An exception occurred while inserting file, file creation will be aborted");
+                throw;
             }
 
             if (file.Children is not null)
             {
                 foreach (var child in file.Children)
                 {
                     var csrsChild = child.ToDynamicsModel();
-                    if (party.SsgPartygender == _male)
-                    {
-                        csrsChild.SsgChildsFatherODataBind = _dynamicsClient.GetEntityURI("ssg_csrsparties", party.SsgCsrspartyid);
-                    }
-                    else if (party.SsgPartygender == _female)
-                    {
-                        csrsChild.SsgChildsMotherODataBind = _dynamicsClient.GetEntityURI("ssg_csrsparties", party.SsgCsrspartyid);
-                    }
 
-                    if (otherParty is not null && otherParty.SsgPartygender == _male)
+                    if (csrsFile is not null && !string.IsNullOrEmpty(csrsFile.SsgCsrsfileid))
                     {
-                        csrsChild.SsgChildsFatherODataBind = _dynamicsClient.GetEntityURI("ssg_csrsparties", otherParty.SsgCsrspartyid);
+                        csrsChild.SsgFileNumberODataBind = _dynamicsClient.GetEntityURI("ssg_csrsfiles", csrsFile.SsgCsrsfileid);
                     }
-                    else if (otherParty is not null && otherParty.SsgPartygender == _female)
-                    {
-                        csrsChild.SsgChildsMotherODataBind = _dynamicsClient.GetEntityURI("ssg_csrsparties", otherParty.SsgCsrspartyid);
-                    }
-
-                    csrsChild.SsgFileNumberODataBind = _dynamicsClient.GetEntityURI("ssg_csrsfiles", csrsFile.SsgCsrsfileid);
 
                     try
                     {
                         csrsChild = await _dynamicsClient.Ssgcsrschildren.CreateAsync(csrsChild, cancellationToken: cancellationToken);
                     }
                     catch (Exception ex)
                     {
-                        _logger.LogError(ex, "Exception in FileService.CreateChildren");
+                        _logger.LogError(ex, "An exception occurred while inserting child in file {FileId}", csrsFile.SsgCsrsfileid);
```

Comment:
Is there by any chance unintentional mix up of **composite-formatting** (https://docs.microsoft.com/en-us/dotnet/standard/base-types/composite-formatting) versus **string interpolation** (https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated) ?

Thanks!